include/NovelRT/Maths/GeoVector2F.h

```diff
@@ -21,157 +24,789 @@ namespace NovelRT::Maths
+         */
         inline GeoVector2F& operator-=(GeoVector2F other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) =
                 *reinterpret_cast<const glm::vec2*>(this) - *reinterpret_cast<const glm::vec2*>(&other);
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the Hadamard product of this and another GeoVector2F.
+         *
+         * @details
+         * An example of a Hadamard product of two vectors: \f[
+         *      \begin{pmatrix}
+         *      2\\
+         *      3
+         *      \end{pmatrix}
+         *      \circ
+         *      \begin{pmatrix}
+         *      4\\
+         *      5
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      2 * 4\\
+         *      3 * 5
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      8\\
+         *      15
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other vector who's components get multiplied by this vector's components.
+         * @return A GeoVecor2F who's components consist of the Hadamard product of this and another GeoVector2F.
+         */
         inline GeoVector2F operator*=(GeoVector2F other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) =
                 *reinterpret_cast<const glm::vec2*>(this) * *reinterpret_cast<const glm::vec2*>(&other);
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the quotents of a Hadamard division using this and another
+         * GeoVector2F.
+         *
+         * @details
+         * An example of a Hadamard division of two vectors: \f[
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      \oslash
+         *      \begin{pmatrix}
+         *      2\\
+         *      3
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      \tfrac{8}{2}\\
+         *      \tfrac{6}{3}
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      4\\
+         *      2
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other vector who's component act as a divisor for this vector's components.
+         * @return A GeoVecor2F who's components consist of the entrywise quotent of this and another GeoVector2F.
+         */
         GeoVector2F operator/=(GeoVector2F other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) =
                 *reinterpret_cast<const glm::vec2*>(this) / *reinterpret_cast<const glm::vec2*>(&other);
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise sum of this and another scalar and applies it to
```

Comment:
construct on operator. Also this is in-place memory overwriting as it returns `T&` to `this`.